mesonbuild/modules/cuda.py

```diff
@@ -320,7 +325,7 @@ def _nvcc_arch_flags(self, cuda_version, cuda_arch_list='Auto', detected=''):
                     'Volta':         (['7.0'],             ['7.0']),
                     'Xavier':        (['7.2'],             []),
                     'Turing':        (['7.5'],             ['7.5']),
-                    'Ampere':        (['8.0'],             ['8.0']),
+                    'Ampere':        (cuda_ampere_tuple,   cuda_ampere_tuple),
```

Comment:
The left side is what binaries it will be compiled for, and the right side is the PTX that will be generated if any. It isn't necessary to generate PTX for both 8.0 and 8.6 because PTX 8.6 is supposed to dominate PTX 8.0.

In the `"Kepler"` [example](https://github.com/mesonbuild/meson/blob/937ff4a334b4e5d3774311630ab6285dec169b27/mesonbuild/modules/cuda.py#L320), as you can see, I used 3.0, 3.5 for binaries and 3.5 for PTX, so that if a later GPU came out, it could at least runtime-compile the PTX of the newer 3.5 ISA.

For the Tegra/Orin/Xavier boards, I translate to binaries without PTX `[]` because realistically nobody will ever build a CUDA binary for these boards without targeting it at one specific microarchitecture.

I'd recommend
```python
cuda_ampere_bin = ['8.0']
cuda_ampere_ptx = ['8.0']
if version_compare(cuda_version, '>=11.1'):
    cuda_ampere_bin += ['8.6']
    cuda_ampere_ptx  = ['8.6']
```
and
```
'Ampere':        (cuda_ampere_bin,     cuda_ampere_ptx),
```

Otherwise, LGTM!

Reply:
done